src/server/hook_func.c

```diff
@@ -982,9 +982,9 @@ mgr_hook_delete(struct batch_request *preq)
 {
 
 
-	hook 		*phook;
-	char		*hookname;
-	char		hook_msg[HOOK_MSG_SIZE];
+	hook *phook = NULL;
+	char hookname[PBS_MAXSVRJOBID + 1] = {0};
```

Comment:
Why a size of PBS_MAXSVRJOBID? Is that not too small? can be hookname be larger than that.

Reply:
@subhasisb I have used PBS_MAXSVRJOBID only for one reason, as source buffer is length of PBS_MAXSVRJOBID.
(see [here](https://github.com/PBSPro/pbspro/blob/master/src/include/batch_request.h#L146))

Reply:
I think it will cause confusion.  It looks like a bug that you are sizing your buffer for a hook name by the max size for a job id.

Reply:
Hey @bhroam, Sorry but I didn't get your point. Are you asking to change buffer size in source buffer? or Are you asking to make new macro with same size as PBS_MAXSVRJOBID and use it?

Reply:
I did a little research and the internal hook code doesn't limit the length of a hook name.  The only limitation is from the batch protocol itself.  When the hook name is set in the hook structure, it uses strdup().  You shouldn't use a fixed length buffer here because if we ever increase the size of the name field in the rq_manage structure, you'll be limiting the size of a hook name.